cmake/EthDependencies.cmake

```diff
@@ -62,3 +62,44 @@ foreach (BOOST_COMPONENT IN LISTS BOOST_COMPONENTS)
 	get_property(LOCATION TARGET Boost::${BOOST_COMPONENT} PROPERTY IMPORTED_LOCATION)
 	message(STATUS "Found Boost::${BOOST_COMPONENT} at ${LOCATION}")
 endforeach()
+
+if(NOT USE_SYSTEM_LIBRARIES)
+	file(GLOB fmtlib_contents "${CMAKE_SOURCE_DIR}/deps/fmtlib/*")
+	file(GLOB nlohmannjson_contents "${CMAKE_SOURCE_DIR}/deps/nlohmann-json/*")
+	file(GLOB rangev3_contents "${CMAKE_SOURCE_DIR}/deps/range-v3/*")
+
+	if (fmtlib_contents AND nlohmannjson_contents AND rangev3_contents)
+		message(STATUS "git submodules seem to be already initialized: nothing to do.")
+	else()
+		message(STATUS "git submodules seem not to be initialized: implicitly executing 'git submodule update --init'.")
+		find_package(Git)
+		if (NOT Git_FOUND)
+			message(FATAL_ERROR "Failed to initialize submodules: 'git' command not found.")
+		endif()
+		execute_process(
+			COMMAND git submodule update --init
+			WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+			RESULT_VARIABLE result
+		)
+		if(NOT result EQUAL 0)
+			message(FATAL_ERROR "Failed to initialize submodules: 'git submodule update --init' failed.")
+		endif()
+	endif()
+
+	if (NOT WIN32)
+		file(GLOB potential_submodules "${CMAKE_SOURCE_DIR}/deps/*")
+		set(expected_potential_submodule_count 3)
+		set(potential_submodule_count 0)
+		foreach(ITEM ${potential_submodules})
+			if(IS_DIRECTORY ${ITEM})
+				math(EXPR potential_submodule_count "${potential_submodule_count} + 1")
+			endif()
+		endforeach()
+		if (NOT ${potential_submodule_count} EQUAL ${expected_potential_submodule_count})
+			message(FATAL_ERROR
+				"There are more directories (potential submodules) found (${potential_submodule_count}) in '${CMAKE_SOURCE_DIR}/deps' than expected (${expected_potential_submodule_count}). "
+				"This is just a reminder to potentially update e.g. '${CMAKE_SOURCE_DIR}/cmake/EthDependencies.cmake'."
```

Comment:
"just a reminder to potentially update" but a `FATAL_ERROR` message :D

Reply:
I mean, for this to be actually useful, a `FATAL_ERROR` seemed like a better choice. Otherwise I'm not sure anyone will notice this warning. It's just the message did not really fit sucn an error. IMO it should have just been reworded, not changed into warning :)

But I also think we'd be fine without this check. It's a lot of code for an unlikely case where someone forgets to commit a new submodule, which they'll immediately notice due to CI failures and fix.

What I was more worried about earlier was instead a situation where someone properly adds and pushes a module but then everyone who pulls that code gets an error because cmake assumes that it's already initialized because it only checks `fmtlib`. But you already fixed that by extending the `if`. The warning wasn't really necessary.